### DISCUSSION TITLE
Join on sub query

### DISCUSSION THREAD

**Discussion by evanxg852000** [diesel-rs/diesel#2921] [Q&A]
Can diesel allow `left_join` on sub query?
Something like ...
```rust
let stmt =  users::table
            .left_join( votes
                 .filter(votes.name.eq("v"))
                 on(votes.id.eq(users.id))
            );
            .select(votes::user_name);
```

**Comment by weiznich** [comment]
If someone figures out the technical details of such a design this should be possible. After that it needs to be implemented. Any help there is welcome.
